src/controllers/booking.controller.ts

```diff
@@ -0,0 +1,170 @@
+import * as Koa from 'koa'
+import * as HttpStatus from 'http-status-codes'
+import {getRepository, Repository} from 'typeorm'
+import Tables from "../models/tables.entity";
+import Booking from "../models/booking.entity";
+import TimetobookEntity from "../models/timetobook.entity";
+import BookedUsersEntity from "../models/bookedusers.entity";
+
+
+export default class BookingController {
+    static async getBooking(ctx: Koa.Context) {
+        const timeArray: Repository<TimetobookEntity> = getRepository(TimetobookEntity)
+        let time = await timeArray.find()
+        const bookingRepo: Repository<Booking> = getRepository(Booking)
+        let booked: Booking[] = await bookingRepo.find({
+            where: {
+                date: ctx.request.query.date
+            }
+        })
+        const tables: Repository<Tables> = getRepository(Tables)
+        let amounts = await tables.find()
+        let newArr: string[]
+        let notAllowedArr: Booking[] = []
+        newArr = [...time.map((el) => {
+            return el.avalibletime
+        })]
+        switch (ctx.request.query.index) {
+            case '0':
```

Comment:
A lot of repetitive and, I would say, not fully understandable code. Let's move it into a separate function, util, etc., so that there isn't a lot of logic here
Btw, whats going on here ?

Reply:
I'm not sure if this is possible with your current approach, but it would be better to store this data in a db. And based on the data received, run the query and see if it is available instead of these incomprehensible switch cases